src/Action.hs

```diff
@@ -69,13 +77,35 @@ import Test.QuickCheck (arbitrary, generate)
 
 -- | User representation during an action
 -- FIXME: Figure out which information is needed here.
-data UserState
-    = UserLoggedOut
-    | UserLoggedIn { _username :: UserLogin, _sessionCookie :: ST }
+data UserState = UserState
+    { _usSessionToken :: Maybe ThentosSessionToken
```

Comment:
The session token is used to make CSRF tokens session specific. The ThentosSessionToken is no more than a nonce (in crypto terms).